src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -188,13 +188,31 @@ private void setUserInfoInThreadContext(User user, Set<String> mappedRoles) {
             joiner.add(String.join(",", user.getRoles()));
             joiner.add(String.join(",", Sets.union(user.getSecurityRoles(), mappedRoles)));
             String requestedTenant = user.getRequestedTenant();
-            if (!Strings.isNullOrEmpty(requestedTenant)) {
```

Comment:
why was this null check removed?

Reply:
The parsing of this string in the downstream plugins happens in a weird way. 
In case of empty requested Tenant(Global Tenant) 

if we don't add anything for requestedTenant.
        The string generated would look like this `myuser|Roles|securityRoles|No`
If we add irrespective
       The string generated would look like this `myuser|Roles|securityRoles||No`

The extra pipe would help downstream plugins not to mistake tenancyAccessRole with tenantName.